Resources/doc/annotations.rst

```diff
@@ -252,3 +252,76 @@ Automatically, registers the given class as a listener with the Doctrine ORM or
 
     ``@DoctrineListener`` implies ``@Service`` if not explicitly defined.    
 
+
+Custom Annotations
+~~~~~~~~~~~~~~~~~~
+Build your own annotations and register them
+
+.. code-block :: php
+
+    <?php
+
+    namespace AppBundle\Annotation;
+
+    use JMS\DiExtraBundle\Annotation\DoctrineListener;
+    use JMS\DiExtraBundle\Metadata\ClassMetadata;
+
+    /**
+     * @Annotation
+     */
+    class Special implements MetadataProcessorInterface
+    {
+        public function processMetadata(ClassMetadata $metadata)
+        {}
+    }
+
+register your annotations, either inside your Bundle configuration or the app config,
+the container parameter jms_di_extra.annotation_patterns is an array with all namespaces.
+
+.. configuration-block ::
+    .. code-block :: yml
+
+        jms_di_extra:
+            annotation_patterns:
+                - "AppBundle\Annotation"
+
+    .. code-block :: php
+
+        <?php
+
+        namespace AppBundle;
+
+        use Symfony\Component\DependencyInjection\ContainerBuilder;
+        use Symfony\Component\HttpKernel\Bundle\Bundle;
+
+        class AppBundle extends Bundle
+        {
+            public function build(ContainerBuilder $container)
+            {
+                if ($container->getParameterBag()->has('jms_di_extra.annotation_patterns')) {
+                    $patterns = $container->getParameter('jms_di_extra.annotation_patterns');
+                } else {
+                    $patterns = array();
+                }
+
+                $patterns[] = 'AppBundle\Annotation';
+
+                $container->setParameter('jms_di_extra.annotation_patterns', $patterns);
```

Comment:
:/ now we have a loop

-> we shall not have a function to add new namespaces on the bundle and should handle extensions through container parameters
-> we shall not overwrite the container parameter.

(both solutions are not perfect, but I do not see another solution)

do you have a suggestion of how to add those patterns?